writer.go

```diff
@@ -0,0 +1,166 @@
+package sequencefile
+
+import (
+	"bytes"
+	"compress/gzip"
+	"crypto/rand"
+	"encoding/binary"
+	"io"
+
+	"github.com/golang/snappy"
+)
+
+type Writer struct {
+	Header        Header
+	writer        io.Writer
+	sinceLastSync int
+	compressor    compressor
+}
+
+func NewWriter(w io.Writer) *Writer {
+	header := Header{
+		Version:        6,
+		KeyClassName:   "org.apache.hadoop.io.BytesWritable",
```

Comment:
I agree. I'm currently operating under the "just modify the underlying `Header` like I am doing in the tests with compression bits.